src/js/13-open-nested-tabs.js

```diff
@@ -25,53 +31,172 @@
+
+    syncGroupIds.forEach(syncGroupId => {
+      // Select all tab groups with this synchronization group ID
+      const tabGroups = document.querySelectorAll(`[data-sync-group-id="${syncGroupId}"]`);
+      tabGroups.forEach(group => {
+        // Deactivate all tabs in this group
+        const tabs = group.querySelectorAll('li.tab');
+        tabs.forEach(tab => {
+          tab.classList.remove('active');
+          tab.setAttribute('aria-selected', 'false');
+          tab.setAttribute('tabindex', '-1');
+        });
+
+        // Hide all tab panels in this group
+        const panels = group.querySelectorAll('.tabpanel');
+        panels.forEach(panel => {
+          panel.classList.add('is-hidden');
+          panel.setAttribute('hidden', '');
+        });
+
+        // Activate the target tab in this group
+        const currentTab = group.querySelector(`#${tabId}`);
+        if (currentTab) {
+          currentTab.classList.add('active');
+          currentTab.setAttribute('aria-selected', 'true');
+          currentTab.setAttribute('tabindex', '0');
+
+          // Show the corresponding tab panel
+          const panelId = `${tabId}--panel`;
+          const targetPanel = group.querySelector(`#${panelId}`);
+          if (targetPanel) {
+            targetPanel.classList.remove('is-hidden');
+            targetPanel.removeAttribute('hidden');
+
+            // Optionally, scroll to the activated tab panel
+            targetPanel.scrollIntoView({ behavior: 'smooth' });
+          }
+        }
+      });
     });
-  }, 300);
 
-  window.addEventListener('DOMContentLoaded', function () {
+    // Update the URL with the active tab ID
     const url = new URL(window.location.href);
-    const tabParam = url.searchParams.get('tab');
+    url.searchParams.set('tab', tabId);
+    window.history.pushState(null, '', url);
 
-    function getClosestSyncGroupId(element) {
-      if (element) {
-        return element.closest('[data-sync-group-id]');
+    // Trigger Prism.js highlighting
+    debouncedHighlightAll();
+  }
+
+  /**
+   * Removes the 'tab' query parameter from the URL if an anchor is present.
+   * Ensures that the hash fragment takes precedence over the 'tab' parameter.
+   */
+  function stripTabParamIfHashPresent(url) {
+    if (url.hash && url.searchParams.has('tab')) {
+      url.searchParams.delete('tab');
+      window.history.replaceState(null, '', url);
+    }
+  }
+
+  /**
+   * Handles deep linking by activating the appropriate tab based on the URL's query parameter or hash fragment.
+   */
+  function handleDeepLinking() {
+    const url = new URL(window.location.href);
+    stripTabParamIfHashPresent(url);
+
+    // Re-parse the URL after potential modification
+    const updatedUrl = new URL(window.location.href);
+    const updatedTabParam = updatedUrl.searchParams.get('tab');
+    const hash = updatedUrl.hash.substring(1); // Remove the '#' character
```

Comment:
Is hash guarantee that will always be on that place?

Reply:
I believe so: https://developer.mozilla.org/en-US/docs/Web/API/URL/hash